django/db/models/fields/__init__.py

```diff
@@ -703,6 +704,9 @@ def contribute_to_class(self, cls, name, private_only=False, virtual_only=NOT_PR
             setattr(cls, 'get_%s_display' % self.name,
                     curry(cls._get_FIELD_display, field=self))
 
+        if self.readonly:
+            cls._meta.has_readonly_fields = True
```

Comment:
Agreed, this is exactly what I proposed 4 minutes after you comment 😄. This code can be removed now as it seems unused.